keras/src/models/functional.py

```diff
@@ -102,6 +102,11 @@ def __new__(cls, *args, **kwargs):
     @tracking.no_automatic_dependency_tracking
     def __init__(self, inputs, outputs, name=None, **kwargs):
         if isinstance(inputs, dict):
+            # This implementation relies on the deterministic order of
+            # dictionary keys
+            # Assumption from Python < 3.7.
+            self._input_names = list(inputs.keys())
+            self._inputs_struct = inputs
```

Comment:
This is already done by `Function.__init__` line 144. In fact `Function.__init__` will overwrite what you set here.